src/graph/planner/ngql/GoPlanner.cpp

```diff
@@ -454,43 +474,40 @@ SubPlan GoPlanner::nStepsPlan(SubPlan& startVidPlan) {
   loopStepVar_ = qctx->vctx()->anonVarGen()->getVar();
 
   auto* start = StartNode::make(qctx);
-  PlanNode* scan = nullptr;
+  PlanNode* getDst = nullptr;
+
+  PlanNode* loopBody = nullptr;
+  PlanNode* loopDep = startVidPlan.root;
   if (!goCtx_->joinInput && goCtx_->limits.empty()) {
     auto* gd = GetDstBySrc::make(qctx, start, goCtx_->space.id);
     gd->setSrc(goCtx_->from.src);
     gd->setEdgeTypes(buildEdgeTypes());
     gd->setInputVar(goCtx_->vidsVar);
-    gd->setColNames({kDst});
-    scan = gd;
+    gd->setColNames({goCtx_->dstIdColName});
+    auto* dedup = Dedup::make(qctx, gd);
```

Comment:
The result of getDstBySrc operator has been deduplicated, why need dedup operator

Reply:
The operator is only de-duplicated on the storaged side. When graphd merges the results from multiple storaged, it needs to be de-duplicated again.

Reply:
got it